### PR TITLE
refactor borrower book details to one Activity with different actions for each book status

### PR DESCRIPTION
**Pull request by nayancprakash**
### Description of changes
* Refactor ExploreBookDetailsActivity, AcceptedBookDetailsActivity, RequestedBookDetailsActivity and BorrowedBookDetailsActivity to one common activity, BorrowerBookDetailsActivity
* The action button of BorrowerBookDetailsActivity is configured differently based on each possible value of the book's status
* Add `onResume()` handler to MyBookDetailsActivity to update the current book as well as the displayed details and the action button configuration upon resuming of activity (this is necessary for handling MyBookDetailsActivity resumption after accepting a request through ManageRequestsActivity)

### CODE REVIEW

File: src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    private void configureActionButton() {
+        switch (book.getStatus()) {
+            case AVAILABLE:
+                actionButton.setText("Request");
+                actionButton.setOnClickListener(requestBookListener);
+                break;
+            case REQUESTED:
+                // TODO: set button style to OutlinedButton
+                actionButton.setText("Requested");
+                actionButton.setOnClickListener(aVoid -> {});
```
Comment (nayancprakash) [inline src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java:148]: anyone know of a way to programmatically set the button style to `Button.OutlinedButton`?

Comment (KonradStanski) [reply src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java]: ContextThemeWrapper newContext = new ContextThemeWrapper(baseContext, R.style.MyStyle);
button = new Button(newContext);

File: src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    private static final int SCAN_ISBN_TO_RETURN = 101;
+    private static final int SCAN_ISBN_TO_BORROW = 102;
+
+    String isbn;
+    String title;
+    String author;
+    String description;
+    String ownedBy;
+    String status;
+    // TODO: Image image;
```
Comment (kylehennig) [inline src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java:41]: Just as a heads up, you should be able to store, retrieve, and delete Photograph objects to Firebase now.

**Review by kylehennig — APPROVED** [review]

File: src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java (hunk 2 above)
Comment (KonradStanski) [reply src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java:41]: Once this is approved, merged, and tested i'll go through the whole app turning them on everywhere/debuggin

File: src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java (hunk 1 above)
Comment (KonradStanski) [reply src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java]: also https://github.com/airbnb/paris

Comment (KonradStanski) [reply src/app/src/main/java/com/example/bookmark/BorrowerBookDetailsActivity.java]: https://ask.xiaolee.net/questions/1042508

**Pull request merged**